scala-thoughts.md

```diff
@@ -0,0 +1,11 @@
+***What's Easy to do in Scala? What's hard?
+So far I think that I like pretty much everything about Scala. It really incorporates almost all of my favorite features from SML (a languge I initially hated but really really learned to love) while still having OOP and readable error messages (among other improvements). Obviously learning any language is hard and Scala is no exception, but despite my frustrations I really enjoyed my first spin. I think that Scala has a really really steep learning curve. Not an easy language to learn.
+
+***What are your favorite design choices?
+Love the arrows, love the pattern matching, love the emphasis on recursion. When I first learned recursion I was really scared of it but it is truly a beautiful and clean solution to many many problems. I also like how clean the if/else statements are. I also really like the way that it almost feels like java when you are doing OOP stuff but really feels like SML when pattern matching.
```

Comment:
I spent a lot of time in Java last summer, so to me it doesn't feel enough like Java when doing OOP :cry: